### ISSUE TITLE
Rule Change: (fill in)

### ISSUE THREAD

**Issue by adamreisnz** [eslint/eslint#16391]
### What rule do you want to change?

spaced-comment

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new option

### Example code

```js
//Some developers may prefer no space for
//inline comments,

/** 
 * While prefering a space when docblock comments
 * are used.
 */
```

### What does the rule currently do for this code?

Currently, the rules enforces it either one way or the other, but you can't have a different rule for inline vs docblock.

### What will the rule do after it's changed?

I'd like to see us allow to specify whether `always` vs `never` applies to inline comments, docblock comments, or all comments. 

The default can be all comments, so that it doesn't break existing rulesets.

Example:

```
  'spaced-comment': ['error', {
      'line': {
        'style': 'never',
        'markers': ['/'],
      },
      'block': {
        'style': 'always',
        'markers': [' '],
        'balanced': true,
      },
    }],
```

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @adamreisnz, thanks for the issue! 

Per our rules policies, [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates), so we are not adding new options to the `spaced-comment` rule. You can always copy the existing rule into a custom rule and make it behave the way you want.

**Comment by adamreisnz** [comment]
Thanks, can you clarify why? The blurb you linked only explains the
outcome, but not the reason why. Thanks.

**Comment by mdjermanovic** [comment]
The reasons for this policy are explained in this blog post:

https://eslint.org/blog/2020/05/changes-to-rules-policies/
